drivers/hopwa_caper/app/models/hopwa_caper/enrollment.rb

```diff
@@ -0,0 +1,130 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HopwaCaper
+  class Enrollment < GrdaWarehouseBase
+    include SqlHelper
+    self.table_name = 'hopwa_caper_enrollments'
+
+    has_many :services, class_name: 'HopwaCaper::Service', foreign_key: [:hud_enrollment_id, :data_source_id, :report_instance_id], primary_key: [:hud_enrollment_id, :data_source_id, :report_instance_id]
+
+    def self.as_report_members
+      all.map do |record|
+        ::HudReports::UniverseMember.new(
+          universe_membership_type: sti_name,
+          universe_membership_id: record.id,
+        )
+      end
+    end
+
+    scope :overlapping_range, ->(start_date:, end_date:) {
+      where('entry_date <= :end_date AND (exit_date >= :start_date OR exit_date IS NULL)', start_date: start_date, end_date: end_date)
+    }
+
+    scope :latest_by_personal_id, -> {
+      select('DISTINCT ON (hopwa_caper_enrollments.data_source_id, hopwa_caper_enrollments.hud_personal_id) *').order(data_source_id: :asc, hud_personal_id: :asc, entry_date: :desc, id: :desc)
```

Comment:
We also have the [ActiveRecordDistinctOn](https://github.com/alecdotninja/active_record_distinct_on) gem.   I believe this can be written as:
```
distinct_on(:data_rouce_id).order(data_source_id: :asc, hud_personal_id: :asc, entry_date: :desc, id: :desc)
```

Reply:
on interesting, thanks for pointing out that gem. Yes, client_id would be a simpler discriminator here